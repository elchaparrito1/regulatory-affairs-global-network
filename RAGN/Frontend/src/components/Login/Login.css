body,
html {
  height: 100%;
  margin: 0;
}

.bg-image {
  background-image: url('../../images/background-home.jpg');
  height: 100%;
  position: fixed;
  top: 0;
  width: 100%;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}

.form-radio {
  /* display: inline-block;
  position: relative;
  background-color: #939393;
  color: white;
  top: 7px;
  height: 20px;
  width: 20px;
  border-radius: 50px;
  cursor: pointer;
  margin-right: 7px; */
}

.form-radio:checked::before {
  position: absolute;
  font: 13px/1 'Open Sans', sans-serif;
  left: 13px;
  top: 5px;
  content: '\02143';
  transform: rotate(40deg);
}

.form-radio:focus {
  outline-color: #312b7f;
}

.form-radio:hover {
  outline-color: #312b7f;
}

.error {
  border-color: red;
}

.no-error {
  border-color: none;
}

/* .form-radio:checked {
  background-color: #312b7f;
} */

/* .input-style {
  font-size: 15px;
  padding: 8px 15px;
  background-color: #939393;
  color: #edeae7;
  border: solid 1px green;
  border-color: #939393;
}

.input-style:hover {
  cursor: pointer;
  background-color: #312b7f;
  color: white;
}

.input-style:active {
  border: solid 1px green;
} */
