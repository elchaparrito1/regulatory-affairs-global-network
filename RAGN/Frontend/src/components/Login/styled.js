import styled from 'styled-components';

export const Row = styled.div`
  &::after {
    content: '';
    clear: both;
    display: table;
  }
`;

function getWidthString(span) {
  if (!span) return;

  let width = (span / 12) * 100;
  return `width ${width}%;`;
}

export const Column = styled.div`
  float: left;
  ${({ xs }) => (xs ? getWidthString(xs) : 'width: 100%')};

  @media only screen and (min-width: 768px) {
    ${({ sm }) => sm && getWidthString(sm)};
  }

  @media only screen and (min-width: 992px) {
    ${({ md }) => md && getWidthString(md)};
  }

  @media only screen and (min-width: 1200px) {
    ${({ lg }) => lg && getWidthString(lg)};
  }
`;

export const Box = styled.div`
  margin: 0 auto;
  background-color: #edeae7;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  height: 450px;
  width: 400px;
  padding: 0 30px 0 30px;

  .input-style {
    font-size: 15px;
    padding: 8px 15px;
    background-color: #939393;
    outline: none;
    color: #edeae7;
    float: left;

    :hover {
      cursor: pointer;
      background-color: #312b7f;
      color: white;
      outline: none;
    }

    :active {
      outline: none;
      border: 0;
    }
  }
`;

export const Text = styled.p`
  font-size: 2em;
  text-align: center;
`;

export const TextMessage = styled.p`
  font-size: 0.8em;
  text-align: right;
  color: red;
  margin-top: -15px;
  margin-right: ${props => (props.password ? '40px' : '5px')};
`;

export const LoginMessage = styled(TextMessage)`
  color: #e68a00;
  text-align: left;
  font-size: 1em;
`;

export const Input = styled.input`
  margin: 0 auto;
  font-size: 1em;
  height: 30px;
  width: 99%;
  margin-bottom: 15px;

  background: transparent;
  border: none;
  border-bottom: 1px solid #000000;

  ::placeholder {
    padding: 5px;
  }

  :focus {
    ::placeholder {
      color: #312b7f;
    }
    color: #312b7f;
    outline: none;
    border-bottom: 1px solid #312b7f;
  }
`;

export const Label = styled.label`
  text-align: center;
`;

export const Button = styled.button`
  font-size: 15px;
  padding: 8px 15px;
  background-color: #939393;
  outline-color: #312b7f;
  color: #edeae7;

  :hover {
    cursor: pointer;
    background-color: #312b7f;
    color: white;
    outline-color: #312b7f;
  }

  :active {
    outline-color: #312b7f;
    border: 0;
  }
`;

export const TextSelection = styled.p`
  font-size: 0.9em;
  display: block;
  margin-left: 5px;

  :hover {
    color: #312b7f;
    opacity: 1;
    cursor: pointer;
  }

  :hover::after {
    width: 100%;
    outline: none;
  }
`;

export const Icon = styled.img`
  width: 30px;
  height: 30px;
  outline: none;

  :hover {
    cursor: pointer;
    outline: none;
  }
`;

export const P = styled.p`
  font-size: 0.9em;
  float: right;
  margin-top: 10px;

  :hover {
    cursor: pointer;
    color: #312b7f;
  }
`;
