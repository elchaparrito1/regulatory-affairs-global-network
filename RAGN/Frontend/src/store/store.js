import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import reducer from '../reducers/root';
import { composeWithDevTools } from 'redux-devtools-extension';

const composeEnhancers = composeWithDevTools({
  // Specify custom devTools options
});

const store = preloadedState =>
  createStore(
    reducer,
    preloadedState,
    composeEnhancers(applyMiddleware(thunk))
  );

export default store;
