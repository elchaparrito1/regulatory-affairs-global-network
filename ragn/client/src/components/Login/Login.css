.form-radio
{
     -webkit-appearance: none;
     -moz-appearance: none;
     appearance: none;
     display: inline-block;
     position: relative;
     background-color: #939393;
     color: white;
     top: 7px;
     height: 20px;
     width: 20px;
     border: 0;
     border-radius: 50px;
     cursor: pointer;     
     margin-right: 7px;
     outline: none;
}

.form-radio:checked::before
{
     position: absolute;
     font: 13px/1 'Open Sans', sans-serif;
     left: 6px;
     top: 3px;
     content: '\02143';
     transform: rotate(40deg);
}

.form-radio:hover
{
     background-color: #f7f7f7;
}

.form-radio:checked
{
     background-color: #312b7f;
}
