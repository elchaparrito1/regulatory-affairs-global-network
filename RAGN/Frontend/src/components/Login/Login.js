import React, { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { login } from '../../actions/session';
import ForgotPassword from '../Forms/ForgotPassword/ForgotPassword';
import {
  Row,
  Column,
  Text,
  Input,
  Label,
  Box,
  TextMessage,
  LoginMessage,
  Icon,
  Button
} from './styled';
import ShowImg from '../../images/show.png';
import HideImg from '../../images/hide.png';
import './Login.css';

const Login = () => {
  const errors = useSelector(state => state.errors);
  const dispatch = useDispatch();

  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [userType, setUserType] = useState('customer');
  const [emailError, setEmailError] = useState('');
  const [passwordError, setPasswordError] = useState('');
  const [emailColor, setEmailColor] = useState('');
  const [passwordColor, setPasswordColor] = useState('');
  const [errorMessage, setErrorMessage] = useState(errors);
  const [showPassword, setShowPassword] = useState(false);

  useEffect(() => {
    setErrorMessage(errors);
  }, [errors]);

  // Method for handling login and firing off action
  const handleSubmit = e => {
    e.preventDefault();
    const user = {
      userType: userType,
      email: email,
      password: password
    };
    dispatch(login(user));
  };

  // Method to check that email has been entered in.
  const checkEmail = () => {
    let color = email === '' ? 'red' : '';
    let error = email === '' ? 'blank' : '';
    setEmailColor(color);
    setEmailError(error);
    return email === '' ? false : true;
  };

  // Method to check that password has been entered in.
  const checkPassword = () => {
    let color = password === '' ? 'red' : '';
    let error = password === '' ? 'blank' : '';
    setPasswordColor(color);
    setPasswordError(error);
    return password === '' ? false : true;
  };

  return (
    <>
      <Row>
        <Column lg="12" md="12" sm="12" xs="12">
          <Box>
            <Row>
              <Column lg="12" md="12" sm="12" xs="12">
                <Text>Log in to RAGN</Text>
              </Column>
            </Row>
            <Row>
              <Column lg="12" md="12" sm="12" xs="12">
                <LoginMessage>{errorMessage}</LoginMessage>
              </Column>
            </Row>
            <form onSubmit={handleSubmit} aria-label="loginInput">
              <Row>
                <Column lg="12" md="12" sm="12" xs="12">
                  <Label>Login type *</Label>
                </Column>
              </Row>
              <div role="radiogroup">
                <Row>
                  <Column lg="12" md="12" sm="12" xs="12">
                    <input
                      type="radio"
                      className="form-radio"
                      value="customer"
                      checked={userType === 'customer'}
                      onChange={event => setUserType(event.target.value)}
                      aria-labelledby="customer"
                    />
                    <label id="customer" htmlFor="customer-selection">
                      {' '}
                      Customer
                    </label>
                  </Column>
                </Row>
              </div>
              <div role="radiogroup">
                <Row>
                  <Column
                    style={{ marginBottom: '15px' }}
                    lg="12"
                    md="12"
                    sm="12"
                    xs="12"
                  >
                    <input
                      type="radio"
                      className="form-radio"
                      value="consultant"
                      checked={userType === 'consultant'}
                      onChange={event => setUserType(event.target.value)}
                      aria-labelledby="consultant"
                    />
                    <label id="consultant" htmlFor="consultant-selection">
                      {' '}
                      Consultant
                    </label>
                  </Column>
                </Row>
              </div>
              <Row>
                <Column lg="12" md="12" sm="12" xs="12">
                  <Label htmlFor="emailInput">Email *</Label>
                </Column>
              </Row>
              <Row>
                <Column lg="12" md="12" sm="12" xs="12">
                  <Input
                    style={{ borderColor: emailColor }}
                    id="email"
                    type="text"
                    aria-label="email-input"
                    placeholder="Your email..."
                    value={email}
                    autoComplete="off"
                    onChange={event => setEmail(event.target.value)}
                    onBlur={checkEmail}
                  />
                </Column>
              </Row>
              {emailError === 'blank' && (
                <Row>
                  <Column lg="12" md="12" sm="12" xs="12">
                    <TextMessage>Valid email required</TextMessage>
                  </Column>
                </Row>
              )}
              <Row>
                <Column lg="12" md="12" sm="12" xs="12">
                  <Label htmlFor="passwordInput">Password *</Label>
                </Column>
              </Row>
              <Row>
                <Column lg="11" md="11" sm="11" xs="11">
                  <Input
                    style={{ borderColor: passwordColor }}
                    id="password"
                    aria-label="password-input"
                    type={showPassword ? 'text' : 'password'}
                    placeholder="Your password..."
                    value={password}
                    autoComplete="off"
                    onChange={event => setPassword(event.target.value)}
                    onBlur={checkPassword}
                  />
                </Column>
                <Column lg="1" md="1" sm="1" xs="1">
                  <button
                    // data-testid="password-button"
                    id="password-btn"
                    type="button"
                    style={{
                      backgroundColor: 'transparent',
                      border: 'none',
                      outlineColor: '#312b7f'
                    }}
                    aria-label="show or hide password"
                    onClick={() => setShowPassword(!showPassword)}
                  >
                    <Icon
                      src={showPassword ? ShowImg : HideImg}
                      alt=" show/hide password icon"
                    />
                  </button>
                </Column>
              </Row>
              {passwordError === 'blank' && (
                <Row>
                  <Column lg="12" md="12" sm="12" xs="12">
                    <TextMessage password>Password required</TextMessage>
                  </Column>
                </Row>
              )}
              <Row style={{ marginRight: '30px' }}>
                <Column lg="6" md="6" sm="6" xs="6">
                  <Button
                    aria-label="login"
                    data-testid="login-button"
                    type="submit"
                  >
                    Login
                  </Button>
                </Column>
              </Row>
            </form>
            <Row>
              <Column lg="12" md="12" sm="12" xs="12">
                <ForgotPassword />
              </Column>
            </Row>
          </Box>
        </Column>
      </Row>
    </>
  );
};

export default Login;
