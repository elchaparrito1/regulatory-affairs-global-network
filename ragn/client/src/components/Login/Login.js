import React from "react";
import API from "../../utils/API";
import { withRouter } from "react-router-dom";
import BlankModal from "../Modals/BlankModal";
import {
    Row,
    Column,
    Text,
    Input,
    Label,
    Box,
    Button,
    TextMessage,
    Icon,
    P
  } from "./styled";
import "./Login.css";
import warningImg from "../../images/warning.png";

class Login extends React.Component {
    state = {
        email: "",
        password: "",
        type: "customer",
        emailError: "",
        passwordError: "",
        typeColor: "",
        emailColor: "",
        passwordColor: "",
        isOpen: false
    };

    // Method for changing state of input fields
    handleInputChange = (name) => (event) => {
        this.setState({
            [name]: event.target.value
        });
    };

    // Method for changing state of radio buttons
    handleChange = (event) => {
        this.setState({
          type: event.target.value
        });
    };

    // Method to check that email has been entered in.
    checkEmail = () => {
        if (this.validateEmail(this.state.email) === false) {
            this.setState({
                emailError: "blank",
                emailColor: "red"
            });
            return false;
        } else {
            this.setState({
                emailError: "",
                emailColor: ""
            });
            return true;
        }
    };

    // Method to check that password has been entered in.
    checkPassword = () => {
        if (this.state.password === "") {
            this.setState({
                passwordError: "blank",
                passwordColor: "red"
            });
            return false;
        } else {
            this.setState({
                passwordError: "",
                passwordColor: ""
            });
            return true;
        }
    };

    // Method to validate email entry.
    validateEmail = () => {
        const { email } = this.state;
        // eslint-disable-next-line
        const regex = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
        // console.log(regex.test(email));
        return regex.test(email);
    };

    // Method for customer/consultant login and input verification. Code is broken down to follow two separate paths based on type selection by user.
    loginSubmit = event => {
        const loginData = {
          email: this.state.email,
          password: this.state.password,
          type: this.state.type
        };
        event.preventDefault();
        this.checkEmail();
        this.checkPassword();
        if (loginData.type === "customer" && loginData.email !== "" && loginData.password !== "") {
            API.customerLogin(loginData)
            .then(response => {
              console.log("response", response);
              if (response.data === "invalid") {
                this.setState({
                    isOpen: true,
                    typeColor: "solid 1px red",
                    emailColor: "red",
                    passwordColor: "red"
                });
              }
              else {
                this.setState({
                  email: "",
                  password: "",
                });
                this.props.history.push({
                  pathname: "/customer",
                  state: { detail: response.data }
                });
                }
            })
            .catch(err => console.log(err));
        } else if (loginData.type === "consultant" && loginData.email !== "" && loginData.password !== "") {
            API.consultantLogin(loginData)
            .then(response => {
              console.log("response", response);
              if (response.data === "invalid") {
                this.setState({
                    isOpen: true,
                    typeColor: "solid 1px red",
                    emailColor: "red",
                    passwordColor: "red"
                });
              }
              else {
                this.setState({
                  email: "",
                  password: ""
                });
                this.props.history.push({
                  pathname: "/consultant",
                  state: { detail: response.data }
                });
                }
            })
            .catch(err => console.log(err));
        }
    };

    // The below methods are used specifically for the modal(s)
    handleModal = () => {
        if(!this.state.isOpen) {
            this.setState({
                isOpen: true
            });
       } else if (this.state.isOpen) {
           this.setState({
                isOpen: false
           });
       }
    };

    render() {
        console.log(this.state.type);
        return (
            <div>
                <BlankModal 
                    isOpen={this.state.isOpen}
                    handleModal={this.handleModal}
                    body={
                        <div>
                            <Row>
                                <Column lg="12" md="12" sm="12" xs="12">
                                    <P><span><Icon src={warningImg}/></span> Login Failed</P>
                                </Column>
                            </Row>
                            <Row>
                                <Column lg="12" md="12" sm="12" xs="12">
                                    <P>Please check credentials used</P>
                                </Column>
                            </Row>
                        </div>
                        }
                    footerMethod={this.handleModal}
                    footer="Close"
                />
                <Row>
                    <Column lg="12" md="12" sm="12" xs="12">
                        <Box>
                            <Row>
                                <Column lg="12" md="12" sm="12" xs="12">
                                    <Text>Log in to RAGN</Text>
                                </Column>
                            </Row>
                        <form>
                            <Row>
                                <Column lg="12" md="12" sm="12" xs="12">
                                    <Label>Login type *</Label>
                                </Column>
                            </Row>
                        <div style={{border: this.state.typeColor}}>
                            <Row>
                                <Column lg="12" md="12" sm="12" xs="12">
                                    <label>
                                        <input
                                            type="radio"
                                            className="form-radio"
                                            value="customer"
                                            checked={this.state.type === "customer"}
                                            onChange={this.handleChange}
                                        />
                                        <span> Customer</span>
                                    </label>
                                </Column>
                            </Row>
                            <Row>
                                <Column style={{marginBottom: "15px"}} lg="12" md="12" sm="12" xs="12">
                                    <label>
                                        <input
                                            type="radio"
                                            className="form-radio"
                                            value="consultant"
                                            checked={this.state.type === "consultant"}
                                            onChange={this.handleChange}
                                        />
                                        <span> Consultant</span>
                                    </label>
                                </Column>
                            </Row>
                        </div>
                            {this.state.typeError === "blank" && (
                            <Row>
                                <Column lg="12" md="12" sm="12" xs="12">
                                    <TextMessage>Choose account type</TextMessage>
                                </Column>
                            </Row>)}
                            <Row>
                                <Column lg="12" md="12" sm="12" xs="12">
                                    <Label>Email *</Label>
                                </Column>
                            </Row>
                            <Row>
                                <Column lg="12" md="12" sm="12" xs="12">
                                    <Input
                                      style={{borderColor: this.state.emailColor}}
                                      id="email"
                                      type="text"
                                      placeholder="Your email..."
                                      value={this.state.email}
                                      autoComplete="off"
                                      onChange={this.handleInputChange("email")}
                                      onBlur={this.checkEmail}
                                    />
                                </Column>
                            </Row>
                            {this.state.emailError === "blank" && (
                            <Row>
                                <Column lg="12" md="12" sm="12" xs="12">
                                    <TextMessage>Valid email required</TextMessage>
                                </Column>
                            </Row>)}    
                            <Row>
                                <Column lg="12" md="12" sm="12" xs="12">
                                    <Label>Password *</Label>
                                </Column>
                            </Row>
                            <Row>
                                <Column lg="12" md="12" sm="12" xs="12">
                                    <Input
                                      style={{borderColor: this.state.passwordColor}}
                                      id="password"
                                      type="password"
                                      placeholder="Your password..."
                                      value={this.state.password}
                                      autoComplete="off"
                                      onChange={this.handleInputChange("password")}
                                      onBlur={this.checkPassword}
                                    />
                                </Column>
                            </Row>
                            {this.state.passwordError === "blank" && (
                            <Row>
                                <Column lg="12" md="12" sm="12" xs="12">
                                    <TextMessage>Password required</TextMessage>
                                </Column>
                            </Row>)}
                                <Row>
                                    <Column lg="12" md="12" sm="12" xs="12">
                                        <Button style={{float: "right"}} onClick={this.loginSubmit}>Login</Button>
                                    </Column>
                                </Row>
                            </form>
                        </Box>
                    </Column>
                </Row>
            </div>
        )
    }
}

export default withRouter(Login);