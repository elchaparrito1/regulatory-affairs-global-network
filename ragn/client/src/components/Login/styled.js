import styled from "styled-components";

export const Row = styled.div`
    &::after {
        content: "";
        clear: both;
        display: table;
    }
`;

function getWidthString(span) {
    if (!span) return;

    let width = span / 12 * 100;
    return `width ${width}%;`;
}

export const Column = styled.div`
    float: left;
    ${({ xs }) => (xs ? getWidthString(xs) : "width: 100%")};

    @media only screen and (min-width: 768px) {
        ${({ sm }) => sm && getWidthString(sm)};
    }

    @media only screen and (min-width: 992px) {
        ${({ md }) => md && getWidthString(md)};
    }

    @media only screen and (min-width: 1200px) {
        ${({ lg }) => lg && getWidthString(lg)};
    }

`;

export const Box = styled.div`
    margin: 0 auto;
    background-color: #edeae7;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
    height: 400px;
    width: 400px;
    padding: 0 30px 0 30px;
`;

export const Text = styled.p`
    font-size: 2em;
    font-family: 'Montserrat', sans-serif;
    text-align: center;
`;

export const TextMessage = styled.p`
    font-size: 0.8em;
    font-family: 'Montserrat', sans-serif;
    text-align: right;
    color: red;
    margin-top: -15px;
`;

export const Input = styled.input`
    font-family: 'Montserrat', sans-serif;
    margin: 0 auto;
    font-size: 1em;
    height: 30px;
    width: 99%;
    margin-bottom: 15px;

    background: transparent;
    border: none;
    border-bottom: 1px solid #000000;
    
    
    :focus {
        border-color: #312b7f;
        outline: 0 none;
    }
`;

export const Label = styled.label`
    font-family: 'Montserrat', sans-serif;
    text-align: center;
`;

export const Button = styled.button`
    font-family: 'Montserrat', sans-serif;
    font-size: 15px;
    padding: 8px 15px;
    background-color: #939393;
    outline: none;
    color: #edeae7;
    
    :hover {
        cursor: pointer;
        background-color: #312b7f;
        color: white;
        outline: none;
    }

    :active {
        outline: none;
        border: 0;
    }
`;

export const TextSelection = styled.p`
    font-family: 'Montserrat', sans-serif;
    font-size: 0.9em;
    display: block;
    margin-left: 5px;

    :hover {
        color: #312b7f;
        opacity: 1;
        cursor: pointer;
    }

    :hover::after {
        width: 100%;
        outline: none;
    }

`;

export const Icon = styled.img`
    width: 20px;

    @media only screen and (min-width: 320px) and (max-width: 767px) {
        height: ${props => props.chevron ? "30px" : "90px;"};
        width: ${props => props.chevron ? "30px" : "90px;"};
    }

    @media only screen and (min-width: 767px) and (max-width: 1024px) {
        height: ${props => props.chevron ? "40px" : "100px;"};
        width: ${props => props.chevron ? "40px" : "100px;"};
    }
`;

export const P = styled.p`
    font-family: 'Montserrat', sans-serif;
    font-size: 20px;
    color: ${props => props.error ? "red" : "black"}
`;

