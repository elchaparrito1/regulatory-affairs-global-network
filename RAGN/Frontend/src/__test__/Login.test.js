import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Login from '../components/Login/Login';
import { Provider } from 'react-redux';
import store from '../store/store';

// test.afterEach(cleanup);

describe('test cases for login', () => {
  const renderComponent = () =>
    render(
      <Provider store={store()}>
        <Login />
      </Provider>
    );

  test('can render with redux with defaults', () => {
    const { getByLabelText } = renderComponent();
    fireEvent.click(getByLabelText(/show or hide password/i));
  });

  test('allows user to login successfully', () => {
    const { getByLabelText } = renderComponent();
    const userType = getByLabelText(/customer/i);
    const email = getByLabelText(/email-input/i);
    const password = getByLabelText(/password-input/i);
    const loginBtn = getByLabelText(/show or hide password/i);

    fireEvent.change(email, { target: { value: 'test@gmail.com' } });
    fireEvent.change(password, { target: { value: 'Test123#' } });
  });
});
