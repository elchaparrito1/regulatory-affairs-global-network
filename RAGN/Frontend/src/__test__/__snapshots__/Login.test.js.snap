// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected React-Redux Component to test login allows user to login successfully 1`] = `
<div
  className="sc-chPdSV jkEqUC"
>
  <div
    className="sc-kgoBCf hdLEOP"
  >
    <div
      className="sc-kGXeez dPIuWb"
    >
      <div
        className="sc-chPdSV jkEqUC"
      >
        <div
          className="sc-kgoBCf hdLEOP"
        >
          <p
            className="sc-kpOJdX fJKtvF"
          >
            Log in to RAGN
          </p>
        </div>
      </div>
      <div
        className="sc-chPdSV jkEqUC"
      >
        <div
          className="sc-kgoBCf hdLEOP"
        >
          <p
            className="sc-dxgOiQ sc-ckVGcZ imVjdu"
          />
        </div>
      </div>
      <form
        aria-label="loginInput"
        onSubmit={[Function]}
      >
        <div
          className="sc-chPdSV jkEqUC"
        >
          <div
            className="sc-kgoBCf hdLEOP"
          >
            <label
              className="sc-eNQAEJ hoEBqV"
            >
              Login type *
            </label>
          </div>
        </div>
        <div
          role="radiogroup"
        >
          <div
            className="sc-chPdSV jkEqUC"
          >
            <div
              className="sc-kgoBCf hdLEOP"
            >
              <input
                aria-labelledby="customer"
                checked={true}
                className="form-radio"
                onChange={[Function]}
                type="radio"
                value="customer"
              />
              <label
                htmlFor="customer-selection"
                id="customer"
              >
                 
                Customer
              </label>
            </div>
          </div>
        </div>
        <div
          role="radiogroup"
        >
          <div
            className="sc-chPdSV jkEqUC"
          >
            <div
              className="sc-kgoBCf hdLEOP"
              style={
                Object {
                  "marginBottom": "15px",
                }
              }
            >
              <input
                aria-labelledby="consultant"
                checked={false}
                className="form-radio"
                onChange={[Function]}
                type="radio"
                value="consultant"
              />
              <label
                htmlFor="consultant-selection"
                id="consultant"
              >
                 
                Consultant
              </label>
            </div>
          </div>
        </div>
        <div
          className="sc-chPdSV jkEqUC"
        >
          <div
            className="sc-kgoBCf hdLEOP"
          >
            <label
              className="sc-eNQAEJ hoEBqV"
              htmlFor="emailInput"
            >
              Email *
            </label>
          </div>
        </div>
        <div
          className="sc-chPdSV jkEqUC"
        >
          <div
            className="sc-kgoBCf hdLEOP"
          >
            <input
              aria-label="email-input"
              autoComplete="off"
              className="sc-jKJlTe wjiUD"
              id="email"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Your email..."
              style={
                Object {
                  "borderColor": "",
                }
              }
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="sc-chPdSV jkEqUC"
        >
          <div
            className="sc-kgoBCf hdLEOP"
          >
            <label
              className="sc-eNQAEJ hoEBqV"
              htmlFor="passwordInput"
            >
              Password *
            </label>
          </div>
        </div>
        <div
          className="sc-chPdSV jkEqUC"
        >
          <div
            className="sc-kgoBCf cAYzjp"
          >
            <input
              aria-label="password-input"
              autoComplete="off"
              className="sc-jKJlTe wjiUD"
              id="password"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Your password..."
              style={
                Object {
                  "borderColor": "",
                }
              }
              type="password"
              value=""
            />
          </div>
          <div
            className="sc-kgoBCf jycTd"
          >
            <button
              aria-label="show or hide password"
              id="password-btn"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "border": "none",
                  "outlineColor": "#312b7f",
                }
              }
              type="button"
            >
              <img
                alt=" show/hide password icon"
                className="sc-kkGfuU hMhJXp"
                src="hide.png"
              />
            </button>
          </div>
        </div>
        <div
          className="sc-chPdSV jkEqUC"
          style={
            Object {
              "marginRight": "30px",
            }
          }
        >
          <div
            className="sc-kgoBCf HaxPP"
          >
            <button
              aria-label="login"
              className="sc-hMqMXs oGbea"
              data-testid="login-button"
              type="submit"
            >
              Login
            </button>
          </div>
        </div>
      </form>
      <div
        className="sc-chPdSV jkEqUC"
      >
        <div
          className="sc-kgoBCf hdLEOP"
        >
          <button
            aria-label="forgot password"
            className="sc-kAzzGY kjeGsl"
            onClick={[Function]}
          >
            Forgot password?
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
